### PR TITLE
do not select from closed ticker

### PR DESCRIPTION
**Pull request by giggsoff**
We try to select from closed ticker.C in loop, which leads to full cpu utilization.
Seems, we can leave timer and check for gotRestarted variable on tick events.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Review by eriknordmark — COMMENTED** [review]
I don't understand why this is an issue. We have other code which selects on a ticker which is stopped (I recall the code in nim.go).

**Review by eriknordmark — APPROVED** [review]
Let it test

File: pkg/pillar/pubsub/socketdriver/subscribe.go [hunk 1]
```diff
@@ -381,26 +381,19 @@ func (s *Subscriber) translate(in <-chan string, out chan<- pubsub.Change) {
...
+			if !ok || !gotRestarted {
+				// skip handling in case of closed or not restarted
+				continue
 			}
 			out <- pubsub.Change{Operation: pubsub.Restart, Key: restartedValue}
 			s.log.Functionf("sent restarted file with %s",
 				restartedValue)
 			gotRestarted = false
 			restartedValue = ""
-			ticker.StopTicker()
```
Comment (eriknordmark) [inline pkg/pillar/pubsub/socketdriver/subscribe.go:403]: Should really restart with the 1000x timer values instead of doing a Stop() here and elsewhere.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
Instead of keeping the ticker running, restart with the 1000x value where we used to Stop.

File: pkg/pillar/pubsub/socketdriver/subscribe.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/pubsub/socketdriver/subscribe.go:403]: Done, it will reduce ticks count.

**Review by eriknordmark — APPROVED** [review]
Great.

**Pull request merged**